### DISCUSSION TITLE
Typescript type for useForm doesn't accept undefined

### DISCUSSION THREAD

**Discussion by peterDuot-stationfive** [react-hook-form/react-hook-form#7825] [Q&A] [answered]
![image](https://user-images.githubusercontent.com/94664894/154012219-ad9e2651-c7f1-427d-9b00-15295cb8efed.png)

I got a question, how do we go around with this typescript error?
Our schema types are generated through codegen.

Thanks.

**Comment by peterDuot-stationfive** [comment, marked as answer]
nevermind, I just added `!` to the parameteres of my mutation so codegen will create a types that is non-nullable.

**Reply by Moshyfawn** [reply]
`FieldValues` is `Record<string, any>`, so the value should be of type object, which shouldn't be `undefined`. `!` does help but it's just the way to tell your TS compiler to STFU. Basically, your form state can't be `undefined`, only `{}`.

**Reply by peterDuot-stationfive** [reply]
Yes, I'm still quite new with typescript. So I'm discovering a lot of things.

**Reply by peterDuot-stationfive** [reply]
Thank you for responding :)
